simulationdataschema/model_system_review.py

```diff
@@ -0,0 +1,864 @@
+        symbols:
+            - https://aflowlib.org/prototype-encyclopedia/chemical_symbols.html
+        """,
+    )
+
+    prototype_aflow_id = Quantity(
+        type=str,
+        description="""
+        The identifier of this structure in the AFLOW encyclopedia of crystallographic prototypes:
+            http://www.aflowlib.org/prototype-encyclopedia/index.html
+        """,
+    )
+
+    atomic_cell_ref = Quantity(
+        type=AtomicCell,
+        description="""
+        Reference to the AtomicCell section that the symmetry refers to.
+        """,
+        a_eln=ELNAnnotation(component="ReferenceEditQuantity"),
+    )
+
+    def resolve_analyzed_atomic_cell(
+        self, symmetry_analyzer: SymmetryAnalyzer, cell_type: str, logger: BoundLogger
+    ) -> Union[AtomicCell, None]:
+        """
+        Resolves the `AtomicCell` section from the `SymmetryAnalyzer` object and the cell_type
+        (primitive or conventional).
+
+        Args:
+            symmetry_analyzer (SymmetryAnalyzer): The `SymmetryAnalyzer` object used to resolve.
+            cell_type (str): The type of cell to resolve, either 'primitive' or 'conventional'.
+
+        Returns:
+            Union[AtomicCell, None]: The resolved `AtomicCell` section or None if the cell_type
+            is not recognized.
+        """
+        if cell_type not in ["primitive", "conventional"]:
+            logger.error(
+                "Cell type not recognized, only 'primitive' and 'conventional' are allowed."
+            )
+            return None
+        wyckoff = getattr(symmetry_analyzer, f"get_wyckoff_letters_{cell_type}")()
+        equivalent_atoms = getattr(
+            symmetry_analyzer, f"get_equivalent_atoms_{cell_type}"
+        )()
+        system = getattr(symmetry_analyzer, f"get_{cell_type}_system")()
+
+        positions = system.get_scaled_positions()
+        cell = system.get_cell()
+        atomic_numbers = system.get_atomic_numbers()
+        labels = system.get_chemical_symbols()
+
+        atomic_cell = AtomicCell(type=cell_type)
+        atomic_cell.lattice_vectors = cell * ureg.angstrom
+        atomic_cell.positions = positions * ureg.angstrom
+        atomic_cell.labels = labels
+        atomic_cell.atomic_numbers = atomic_numbers
+        atomic_cell.wyckoff_letters = wyckoff
+        atomic_cell.equivalent_atoms = equivalent_atoms
+        atomic_cell.get_geometric_space_for_atomic_cell(logger)
+        return atomic_cell
+
+    def resolve_bulk_symmetry(
+        self, original_atomic_cell: AtomicCell, logger: BoundLogger
+    ) -> Tuple[Union[AtomicCell, None], Union[AtomicCell, None]]:
+        """
+        Resolves the symmetry of the material being simulated using MatID and the
+        originally parsed data under original_atomic_cell. It generates two other
+        `AtomicCell` sections (the primitive and standarized cells), as well as populating
+        the `Symmetry` section.
```

Comment:
Does this mean that I cannot set the `Symmetry` from the parser?
Many codes perform their own symmetry analysis. Are we just going to throw that out?

Reply:
No, you can populate it if you want. Then the Symmetry.normalize() will not be run.

Reply:
Normalization always runs, since NOMAD calls it explicitly.

Reply:
Then it is a matter of deciding how to normalize this more consistently. In my case, I defined by hand normalizations, which I don't know how it differs from the fact of having a `SystemNormalizer`, etc. I will talk with Alvin and Lauri about this.

Reply:
> In my case, I defined by hand normalizations

(which means that only the line calling for normalize() under `ModelSystem.normalize()` is ran)

Reply:
Ow, I remember... No, **that's a flawed way of working**: that part of the code should never be modified by any plugin (dev), including those that we produce. The `normalize` interface is a contract between NOMAD and the plugins. You can't modify that contract, but you can refuse to sign it, i.e. we build our own alternative to `normalize`, i.e. `standardize` if you need special control.

If you need more control over normalization, discuss it with Markus S.

Reply:
You see, by modifying the interface one-sidedly, you break the contract and subvert the expectations of all plugin devs.
I can no longer think about our own plugin, because of this, since somebody modified the contract behind my back...

Now imagine that everyone starts doing this...

Reply:
```
if section.m_def.name == 'Simulation' and isinstance(section, EntryData):
                self.normalize_section(section, logger)
                sub_sections = [sub_section for sub_section in section.m_contents()]
                for sub_section in sub_sections:
                    if sub_section.m_def.name == 'ModelSystem':
                        self.normalize_section(sub_section.atomic_cell[0], logger)
                        self.normalize_section(sub_section, logger)
                    elif sub_section.m_def.inherited_sections[2].name == 'TB':
                        self.normalize_section(sub_section, logger)
                        self.normalize_section(sub_section.k_mesh, logger)
```

This entire `if`-section in `metainfo.py` just recognizes our schema.
This logic has to be moved to `ModelSystem` and `TB` respectively, which then call subsections via `standardize`.